search-replace.go

```diff
@@ -155,71 +162,229 @@ func fixLine(line *[]byte, replacements []*Replacement) *[]byte {
 	return line
 }
 
-var serializedStringPrefixRegexp = regexp.MustCompile(`s:(\d+):\\"`)
-
 func fixSerializedContent(line *[]byte, replacements []*Replacement) *[]byte {
-	startIndex := 0
-	for startIndex < len(*line) {
-		Debugf("Start of loop, startIndex: %d\n", startIndex)
-		match := serializedStringPrefixRegexp.FindSubmatchIndex((*line)[startIndex:])
-		if match == nil {
+	index := 0
+
+	var rebuiltLine []byte
+
+	for index < len(*line) {
+		Debugf("Start of loop, index: %d\n", index)
+		linePart := (*line)[index:]
+
+		details, err := parseEscapedData(linePart)
```

Comment:
We then figure out what's within that remaining `linePart` - i.e. where does the content start, the end of it, where the 'next part' should be. `NextPart` here refers to the point where we stop parsing the current part - and that's at the last instance of the string `\";` i.e. in `s:1:\"a\";`. The next part does not necessarily immediately start at another serialized string - `parseEscapedData` has a job to find where the starting point of the next serialized string.